### PR TITLE
Update generator script

### PR DESCRIPTION
**Pull request by Gnimuc**
Clang v0.18 will be adding an experimental C++ environment for parsing C++ source code. 

x-ref: https://github.com/JuliaInterop/Clang.jl/issues/470

### CODE REVIEW

**Comment by Gnimuc** [comment]
I only tested it on mac-aarch64, so if it doesn't work on your machine, please report the issue to Clang.jl. ;)

File: bindings/make.jl [hunk 1]
```diff
@@ -167,8 +167,9 @@ for (julia_version, llvm_version) in julia_llvm
             options["general"]["output_file_path"] = output_file_path
 
             libmlir_header_dir = joinpath(include_dir, "mlir-c")
-            args = Generators.get_default_args()
-            append!(args, ["-I", include_dir, "-x", "c++"])
+            args = Generators.get_default_args(get_triple(); is_cxx=true)
+            push!(args, "-I"*include_dir)
```
Comment (mofeing) [inline bindings/make.jl:171]:
```suggestion
            push!(args, "-I$include_dir")
```

**Review by mofeing — APPROVED** [review]
LGTM, just a small aesthetic change.

**Comment by jumerckx** [comment]
Tested on linux-x86-64 (through WSL) and works great!
I also tested on Julia 1.10 without issues so I think we can get rid of the `Manifest.toml` that was checked in as well?

**Comment by mofeing** [comment]
> Tested on linux-x86-64 (through WSL) and works great! I also tested on Julia 1.10 without issues so I think we can get rid of the `Manifest.toml` that was checked in as well?

I guess so, but we should put a compat bound to `Clang`.

**Comment by mofeing** [comment]
Thanks @Gnimuc for your help!

**Pull request merged**